src/components/organisms/JoinVenueModal/JoinVenueModal.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback } from "react";
+import { Modal } from "react-bootstrap";
+
+import { AnyVenue } from "types/venues";
+
+import { joinVenue } from "utils/url";
+
+import "./JoinVenueModal.scss";
+
+interface JoinVenueModalProps {
+  venueId?: string;
+  venue: AnyVenue;
+  show: boolean;
+  isLoggedIn: boolean;
+  onHide?: () => void;
+}
+
+export const JoinVenueModal: React.FC<JoinVenueModalProps> = ({
+  show,
+  venueId,
+  venue,
+  isLoggedIn,
+  onHide,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, !!venue?.entrance?.length);
+  }, [venue.entrance, venueId]);
+
+  const hideModal = useCallback(() => {
+    onHide && onHide();
+  }, [onHide]);
+
+  return (
+    <Modal show={show} onHide={hideModal}>
+      <Modal.Body>
+        <>
+          <div className="join-venue-title">
+            {isLoggedIn
+              ? `You are now attending ${venue.name}`
+              : "You need an account to join this party"}
+          </div>
+          <button
+            className="btn btn-primary btn-block btn-centered"
+            onClick={handleJoinVenue}
+          >
+            {`Let's go`}
+          </button>
```

Comment:
[NB] This part still feels like it should be what the `VenueJoinButton` component handles, and maybe the 'extra stuff' around checking things and/or popping up this modal/etc that are currently in it perhaps somewhere else?

Basically, having a tiny component that wraps up the button, the text we use for it, the 'enter venue' handler logic, etc, that gets reused wherever we need that button. As right now, this feels duplicated (which isn't a huge deal, but does mean there is a far higher chance of 'fixes'/changes being applied to only one of them, and then them getting 'out of sync' from one another)

Reply:
Also.. the logic above switches the text based on `isLoggedIn`, but this button will always be shown. Is that correct? Feels like I should be given some kind of 'go log in' button or similar? Or maybe just this modal never even popped up, and we go straight to the 'you need to login' or similar?

Reply:
VenueJoinButton should be something like VenueAccessMode or VenueAccessForm because it's responsible to render different button / UI depending on the access in that venue